### PR TITLE
refactor(cardinal): get rid of Create and CreateMany world methods, Replace with functions that use IAbstractComponentType

### PR DESCRIPTION
**Pull request by pyrofolium**
Replace with functions that take in IAbstractComponentType.

Closes: #400
## What is the purpose of the change

To make entity creation rely on IAbstractComponentType

> Add a description of the overall background and high-level changes that this PR introduces
Introduces two functions: `Create(world, IAbstractComponentType)` and `CreateMany(world, int, IAbstractComponentType...)`

Removes two methods: `world.Create(IComponentType)` and world.CreateMany(int, IComponentType)

## Testing

All associated tests have been changed. To work.

### CODE REVIEW

File: cardinal/ecs/component.go [hunk 1]
```diff
@@ -115,6 +115,26 @@ func (c *ComponentType[T]) MustFirst(w *World) entity.ID {
 	return id
 }
 
+func CreateMany(world *World, num int, components ...component.IAbstractComponent) ([]entity.ID, error) {
+	acc := make([]IComponentType, 0, len(components))
+	for _, comp := range components {
+		c, ok := world.GetComponentByName(comp.Name())
```
Comment (jerargus) [inline cardinal/ecs/component.go:121]: This pattern of calling GetComponentByName, and returning an error when it's absent seems to be repeated. I suggest making GetComponentByName return an error, and assigning the "Must register component before creating an entity" to an actual error variable.

File: cardinal/ecs/ecs_test.go [hunk 2]
```diff
@@ -469,25 +473,35 @@ func TestCannotSetComponentThatDoesNotBelongToEntity(t *testing.T) {
...
+
+func (A) Name() string { return "a" }
+func (B) Name() string { return "b" }
+func (C) Name() string { return "c" }
+func (D) Name() string { return "d" }
+
 func TestQueriesAndFiltersWorks(t *testing.T) {
 	world := ecs.NewTestWorld(t)
-	a, b, c, d := ecs.NewComponentType[int]("a"), ecs.NewComponentType[int]("b"), ecs.NewComponentType[int]("c"), ecs.NewComponentType[int]("d")
+	a, b, c, d := ecs.NewComponentType[A]("a"), ecs.NewComponentType[B]("b"), ecs.NewComponentType[C]("c"), ecs.NewComponentType[D]("d")
```
Comment (jerargus) [inline cardinal/ecs/ecs_test.go:496]: This is out of scope for this PR, but now that Name() is defined on these structs, is that first parameter to NewComponentType still required?

**Review by jerargus — APPROVED** [review]
These comments are optional.

File: cardinal/ecs/ecs_test.go (hunk 2 above)
Comment (pyrofolium) [reply cardinal/ecs/ecs_test.go:508]: No. This will be removed in a subsequent PR.

**Pull request merged**